### PR TITLE
Web form pattern added

### PR DESCRIPTION
**Pull request by trevorsaint**
Added new pattern for **web form**.

- Standard form
- Form with errors
- Success page

### CODE REVIEW

**Comment by trevorsaint** [comment]
[Web Form](https://trello.com/c/CvWdNqhI/382-web-form)

File: src/patterns/webform/examples/errors/index.njk [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                    "classes": 'js-inpagelink'
+                },
+                {
+                    "text": 'Enter your name',
+                    "index": true,
+                    "url": '#webform-name-error',
+                    "classes": 'js-inpagelink'
+                },
+                {
+                    "text": 'Enter your email address',
```
Comment (jrbarnes9) [inline src/patterns/webform/examples/errors/index.njk:80]: In eQ census, we only showed the example when an invalid email had been entered: "Enter an email address in a valid format, for example, name@example.com"
For an empty email field we show just...
```suggestion
                    "text": 'Enter your email address',
```

File: src/patterns/webform/examples/errors/index.njk [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            ]
+        })
+    }}
+    
+    {% endcall %}
+    
+    <h1 class="u-fs-xxl u-mt-l">Web form</h1>
+
+    {{ onsRadios({
+        "legend": 'Your country',
```
Comment (jrbarnes9) [inline src/patterns/webform/examples/errors/index.njk:94]: Need content design to confirm, but I'd suggest something like...
```suggestion
        "legend": 'What country do you live in?',
```
...is a better legend. To be consistent with the next fieldset.

File: src/patterns/webform/examples/errors/index.njk [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+            "text": 'Please provide more detail'
+        },
+        "charCheckLimit": {
+            "limit": 5000,
+            "charCountSingular": 'You have {x} character remaining',
+            "charCountPlural": 'You have {x} characters remaining'
+        },
+        "error": {
+            "id": 'webform-detail-error',
+            "text": 'Enter more detail on what you need help with',
```
Comment (jrbarnes9) [inline src/patterns/webform/examples/errors/index.njk:193]:
```suggestion
            "text": 'Enter more detail about what you need help with',
```

File: src/patterns/webform/examples/errors/index.njk [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+            "dsExample": isPatternLib
+        }
+    }) }}
+
+    {{ onsButton({
+        "classes": 'u-mb-m u-mt-l',
+        "text": 'Send message'
+    }) }}
+
+    <p class="u-fs-s">Information about what we do with your personal data is available in our <a href="#0">privacy and data protection statement</a></p>
```
Comment (jrbarnes9) [inline src/patterns/webform/examples/errors/index.njk:236]:
```suggestion
    <p class="u-fs-s">Information about what we do with your personal data is available in our <a href="#0">privacy and data protection statement</a>.</p>
```

File: src/patterns/webform/examples/errors/index.njk [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    }) }}
+
+    {{ onsTextarea({
+        "name": 'webform-detail',
+        "id": 'webform-detail',
+        "label": {
+            "text": 'Please provide more detail'
+        },
+        "charCheckLimit": {
+            "limit": 5000,
```
Comment (jrbarnes9) [inline src/patterns/webform/examples/errors/index.njk:187]: I know this was the character limit on census, but I don't understand why it's so high for a web form.  I'd say 2000 is more than enough.

File: src/patterns/webform/examples/success/index.njk [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        onsPanel({
+            "classes": 'u-mb-m',
+            "type": 'success',
+            "id": 'webform-success',
+            "icon": 'check',
+            "iconSize": 'xl',
+            "body": '<h1>Thank you for contacting us</h1><p>Your message has been sent.</p>'
+        })
+    }}
+    <p>We will respond to you within 4 working days.</p>
```
Comment (jrbarnes9) [inline src/patterns/webform/examples/success/index.njk:35]: This was originally 2, so I would show that in DS. 
```suggestion
    <p>We will respond to you within 2 working days.</p>
```

File: src/patterns/webform/examples/success/index.njk [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        "logoHref": '#0',
+        "titleLogo": 'census-logo-en',
+        "titleLogoHref": '#0',
+        "titleLogoAlt": 'Census 2021'
+    },
+    "footer": {
+    }
+} %}
+{% block main %}
+    {{
```
Comment (jrbarnes9) [inline src/patterns/webform/examples/success/index.njk:25]: I think the success page probably needs CTA as a route onwards for the user.  Either a breadcrumb to 'Home', or a CTA button.  We used 'Done' at the end of all Census RH fulfilment journeys.

File: src/patterns/webform/index.njk [hunk 8]
```diff
@@ -0,0 +1,53 @@
+---
+title: Web form
+group: Ask users for...
+experimental: true
```
Comment (jrbarnes9) [inline src/patterns/webform/index.njk:4]: I think this web form when through a fair bit of user testing? (Check with Alison) so might be ok to publish without the tag.

**Review by jrbarnes9 — COMMENTED** [review]
I would add the documentation before merging this (you might already have planned to do so).  Think now that we have the time, we want to avoid publishing patterns without documentation.

When you do add it, we need to also document all the validation error messages for this form similar to: https://ons-design-system.netlify.app/patterns/dates/#error-messages

File: src/patterns/webform/examples/errors/index.njk (hunk 5 above)
Comment (trevorsaint) [reply src/patterns/webform/examples/errors/index.njk:187]: Agreed. I took this value from what was used in Census2021. Currently live.

File: src/patterns/webform/examples/success/index.njk (hunk 6 above)
Comment (trevorsaint) [reply src/patterns/webform/examples/success/index.njk:35]: Again. The 4 day value is what is in live at the moment.

File: src/patterns/webform/examples/success/index.njk (hunk 7 above)
Comment (trevorsaint) [reply src/patterns/webform/examples/success/index.njk:25]: I will match the 'Done' as in the Census RH fulfilment journeys.

File: src/patterns/webform/index.njk (hunk 8 above)
Comment (trevorsaint) [reply src/patterns/webform/index.njk:4]: I also think Kimberley did some testing on our prototype version of it. I will remove the tag.

File: src/patterns/webform/index.njk [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+# How to use this pattern
+
+The web form pattern includes the following, in this order:
+
+{{
+    onsList({
+        "itemsList": [
+            {
+                "text": 'What country do you live in? (radio selection)'
```
[the remark this answers is not in the document]
Comment (jrbarnes9) [reply src/patterns/webform/index.njk:28]: Probably also need to add details on validation, e.g. make all answers mandatory as you should only ask what you need.

**Review by rmccar — COMMENTED** [review]
Im thinking it should be in a folder named `send-a-web-form` rather than `webform`

**Comment by trevorsaint** [comment]
> Im thinking it should be in a folder named `send-a-web-form` rather than `webform`

Agree. I will update that now.

**Comment by rmccar** [comment]
> > Im thinking it should be in a folder named `send-a-web-form` rather than `webform`
> 
> Agree. I will update that now.
> 
> Looking at other patterns it may be good to follow the same naming pattern. Others do look inconsistent.

Yeah I agree we should standardise it across the repo

**Review by rmccar — APPROVED** [review]
As long as JBs comments are addressed, looks good to me

File: src/patterns/send-a-web-form/index.njk [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+            "linkText": "Design System forum"
+        })
+    }}
+{% endcall %}
+
+## Design System forum
+{{
+    onsExternalLink({
+        "url": "https://github.com/ONSdigital/design-system/discussions/1457",
+        "linkText": "Discuss 'Web form' on GitHub"
```
Comment (rmccar) [inline src/patterns/send-a-web-form/index.njk:87]: Oh also this needs changing and the same will need updating on the github discussion
```suggestion
        "linkText": "Discuss 'Send a web form' on GitHub"
```

Comment (trevorsaint) [reply src/patterns/send-a-web-form/index.njk]: Thanks @rmccar, that has been updated.

**Review by rmccar — DISMISSED** [review]

**Review by rmccar — DISMISSED** [review]

File: src/patterns/send-a-web-form/index.njk [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+## When not to use this pattern
+
+Do not use this pattern if the user needs to speak to someone at the contact centre, for example, if the details about a business need to change. Do not use this pattern for urgent issues that need a fast response, like problems with the service.
+
+## How to use this pattern
+
+To help contact staff deal with web form enquiries, you should provide a set of categories based on the most common user needs from research.
+
+You should only ask the user to provide the contact details with the service needs to handle the request.
```
Comment (jrbarnes9) [inline src/patterns/send-a-web-form/index.njk:24]:
```suggestion
You should only ask the user to provide the contact details the service needs to handle the request.
```

**Review by rmccar — APPROVED** [review]

**Review by jrbarnes9 — APPROVED** [review]

**Pull request merged**